src/AstGen.zig

```diff
@@ -1722,6 +1722,53 @@ fn structInitExpr(
         }
     }
 
+    {
+        var duplicate_names = std.AutoArrayHashMap(u32, ArrayListUnmanaged(Ast.TokenIndex)).init(astgen.arena);
+        defer duplicate_names.deinit();
+        try duplicate_names.ensureTotalCapacity(@intCast(struct_init.ast.fields.len));
+
+        // When there aren't errors, use this to avoid a second iteration.
+        var any_duplicate = false;
+
+        for (struct_init.ast.fields) |field| {
+            const name_token = tree.firstToken(field) - 2;
+            const name_index = try astgen.identAsString(name_token);
+
+            const gop = try duplicate_names.getOrPut(name_index);
+
+            if (gop.found_existing) {
+                try gop.value_ptr.append(astgen.arena, name_token);
+                any_duplicate = true;
+            } else {
+                try gop.value_ptr.append(astgen.arena, name_token);
+            }
+        }
+
+        if (any_duplicate) {
+            var it = duplicate_names.iterator();
+
+            while (it.next()) |entry| {
+                const record = entry.value_ptr.*;
+                if (record.items.len > 1) {
+                    var error_notes = std.ArrayList(u32).init(astgen.arena);
+
+                    for (record.items[1..]) |duplicate| {
```

Comment:
Can you add a test case for multiple redefinitions?

Reply:
I'm sorry what exactly do you mean by this? What is a "redefinition" in this case?

Reply:
Oh, I think I know what you mean 😄 .

You mean a compile_error test for something like:
```zig
    _ = struct {
        a: u32,
        a: u32,
        a: u32,
    };
```

which results in 
```
main.zig:3:9: error: duplicate struct field: 'a'
        a: u32,
        ^
main.zig:4:9: note: other field here
        a: u32,
        ^
main.zig:5:9: note: other field here
        a: u32,
        ^
main.zig:2:9: note: struct declared here
    _ = struct {
        ^~~~~~
```

Reply:
any ideas what i should call the file/test?

Reply:
You could add it to `duplicate_struct_field.zig`.